test: images: add sample-device-plugin to the config

ffromani opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
cleanup: instead of having special handling of the sample device plugin image, add it to the standard list of e2e images.
Note the e2e_node tests will still need some exceptions because they consume container images maintained externally to the project, but this one *is* maintained by the kubernetes project.

#### Which issue(s) this PR fixes:
Fixes N/A

#### Special notes for your reviewer:
At this point in time, only `e2e_node` tests consume this image, but moving it here is very cheap and enables `e2e/node` tests to consume it.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

ffromani committed:
test: images: add sample-device-plugin to the config

`e2e_node` tests need the sample device plugin image.
Instead of adding an exception for this image, add to the
standard set of e2e images in `test/utils/image`.

Note: at this point in time, only `e2e_node` tests consume this image,
but moving it here is very cheap and enables `e2e/node` tests to
consume it.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
/sig node
/sig testing

ffromani commented:
/test

ffromani commented:
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-swap-fedora-serial
/test pull-kubernetes-node-swap-ubuntu-serial
/test pull-kubernetes-node-kubelet-serial-topology-manager
/test pull-kubernetes-e2e-gce-serial

ffromani commented:
```
 W0510 15:55:38.673] 2023/05/10 15:55:38 process.go:155: Step 'bash -c . hack/lib/version.sh && KUBE_ROOT=. kube::version::get_version_vars && echo "${KUBE_GIT_VERSION-}"' finished in 352.108822ms
W0510 15:55:38.682] 2023/05/10 15:55:38 main.go:328: Something went wrong: failed to prepare test environment: error during gcloud compute --project=k8s-jkns-e2e-gce-alpha project-info describe: exit status 1 
```
will retry later

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-containerd

bart0sh commented:
/triage accepted
/priority important-longterm

SergeyKanzhelev commented on test/e2e_node/image_list.go:
```diff
@@ -84,16 +85,12 @@ func updateImageAllowList(ctx context.Context) {
 	} else {
 		e2epod.ImagePrePullList.Insert(gpuDevicePluginImage)
 	}
+	// TODO(1.28+): unify the sample device plugin image, use the latest for everything
```
version reference in TODO is confusing. Do you want to do it in 1.28 in separate PR?

ffromani commented:
/test

ndixita commented:
/assign @swatisehgal

swatisehgal reviewed (commented):
LGTM but want to make sure all the relevant jobs are passing in the CI and we are not introducing regressions:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e-features

I had a concern about upgrading the image version from 1.3 to 1.5 (registry.k8s.io/e2e-test-images/sample-device-plugin:1.5) in [test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml#L35](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml#L35) but double checked in the code and can confirm that if `REGISTER_CONTROL_FILE` is not specified (which is the case in [test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml](test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml)) behavior of 1.3 and 1.5 should be identical. This should also be confirmed with validation in CI as well.

Will formally add LGTM label once these jobs pass in the CI.

ffromani commented:
> LGTM but want to make sure all the relevant jobs are passing in the CI and we are not introducing regressions: /test pull-kubernetes-node-kubelet-serial-containerd /test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial /test pull-kubernetes-cos-cgroupv1-containerd-node-e2e /test pull-kubernetes-cos-cgroupv1-containerd-node-e2e-features
> 
> I had a concern about upgrading the image version from 1.3 to 1.5 (registry.k8s.io/e2e-test-images/sample-device-plugin:1.5) in [test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml#L35](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml#L35) but double checked in the code and can confirm that if `REGISTER_CONTROL_FILE` is not specified (which is the case in [test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml](test/e2e/testing-manifests/sample-device-plugin/sample-device-plugin.yaml)) behavior of 1.3 and 1.5 should be identical. This should also be confirmed with validation in CI as well.
> 
> Will formally add LGTM label once these jobs pass in the CI.

Totally fair, thanks!

bart0sh commented:
/cc
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e-features

marquiz commented:
/retest

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e-features

ffromani commented:
kubernetes-node-e2e-containerd:
```
Test started today at 8:09 AM failed after 3m29s. ([more info](https://prow.k8s.io/spyglass/lens/metadata/iframe?req=%7B%22artifacts%22%3A%5B%22finished.json%22%2C%22podinfo.json%22%2C%22prowjob.json%22%2C%22started.json%22%5D%2C%22index%22%3A0%2C%22src%22%3A%22gs%2Fkubernetes-jenkins%2Fpr-logs%2Fpull%2F117908%2Fpull-kubernetes-node-e2e-containerd%2F1712712129116639232%22%7D&topURL=https%3A//prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/117908/pull-kubernetes-node-e2e-containerd/1712712129116639232&lensIndex=0#))

There are no nodes that your pod can schedule to - check your requests, tolerations, and node selectors (0/85 nodes are available: 1 Insufficient memory, 11 node(s) had untolerated taint {node.kubernetes.io/not-ready: }, 74 Insufficient cpu. preemption: 0/85 nodes are available: 11 Preemption is not helpful for scheduling, 74 No preemption victims found for incoming pod..)
```

will retry later

ffromani commented:
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial

it seems #119882 didn't help as much as I expected :\

ffromani commented:
/test  pull-kubernetes-node-e2e-containerd

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-containerd

ffromani committed:
e2e: node: more feedback about podresources query

add more log to torubleshoot connection failures
to local podresources endpoint after kubelet restart.

Signed-off-by: Francesco Romani <fromani@redhat.com>

bart0sh commented on test/utils/image/manifest.go:
```diff
@@ -266,6 +268,7 @@ func initImageConfigs(list RegistryList) (map[ImageID]Config, map[ImageID]Config
 	configs[Redis] = Config{list.PromoterE2eRegistry, "redis", "5.0.5-3"}
 	configs[RegressionIssue74839] = Config{list.PromoterE2eRegistry, "regression-issue-74839", "1.2"}
 	configs[ResourceConsumer] = Config{list.PromoterE2eRegistry, "resource-consumer", "1.13"}
+	configs[SampleDevicePlugin] = Config{list.PromoterE2eRegistry, "sample-device-plugin", "1.5"}
```
```suggestion
	configs[SampleDevicePlugin] = Config{list.PromoterE2eRegistry, "sample-device-plugin", "1.7"}
```

bart0sh commented:
/remove-lifecycle rotten

bart0sh commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e-features

bart0sh commented:
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2

bart0sh commented:
/retest

bart0sh commented:
/remove-lifecycle stale

ffromani committed:
Update test/utils/image/manifest.go

address review suggestion

Co-authored-by: Ed Bartosh <eduard.bartosh@intel.com>

bart0sh commented on test/e2e_node/image_list.go:
```diff
@@ -89,16 +90,12 @@ func updateImageAllowList(ctx context.Context) {
 	} else {
 		e2epod.ImagePrePullList.Insert(gpuDevicePluginImage)
 	}
+	// TODO: unify the sample device plugin image, use the latest for everything
```
What prevents us from doing it now?

ffromani replied:
Insufficient testing yet. That's my only remaining concern

bart0sh replied:
Would it make sense to create an issue about this?

bart0sh replied:
otherwise the PR looks good to me
/lgtm

ffromani replied:
> Would it make sense to create an issue about this?

yup! done: https://github.com/kubernetes/kubernetes/issues/125227

bart0sh commented:
/retest

bart0sh commented:
/assign @dims @mrunalp 
for approval

BenTheElder commented on test/e2e_node/image_list.go:
```diff
@@ -63,6 +63,7 @@ var NodePrePullImageList = sets.NewString(
 	imageutils.GetE2EImage(imageutils.NodePerfNpbEp),
 	imageutils.GetE2EImage(imageutils.NodePerfNpbIs),
 	imageutils.GetE2EImage(imageutils.Etcd),
+	imageutils.GetE2EImage(imageutils.SampleDevicePlugin),
```
Where possible we're trying to move e2e to have only the agnhost image to simplify maintenance, airgapped testing, etc.

This image looks like something we could fold into agnhost. cc @pohly @aojea

pohly replied:
What is this "SampleDevicePlugin" image? Where is it being maintained?

bart0sh replied:
`SampleDevicePlugin` image is used for testing device plugins. It's maintained here: https://github.com/kubernetes/kubernetes/tree/master/test/images/sample-device-plugin

ffromani replied:
This is the sample device plugin image: https://github.com/kubernetes/kubernetes/tree/v1.30.1/test/images/sample-device-plugin . It has been there since [long long time](https://github.com/kubernetes/kubernetes/commit/0ec6bdd2b5963d5f0c12dd25edd8bf7fb77e1e1a) and it's maintained by SIG-node contributors

ffromani replied:
@BenTheElder do you see the image folding as followup of this PR or as replacement? (or else?)

ffromani commented:
update: I'll file a new PR to consolidate the sample-device-plugin binary into `agnhost` and we we'll move forward from here
